src/models/User.ts

```diff
@@ -0,0 +1,54 @@
+import mongoose, { Schema, Types, Document } from 'mongoose';
+import axios from 'axios';
+import bcrypt from 'bcrypt';
+
+export interface IUser extends Document {
+    firstname: string;
+    lastname: string;
+    username: string;
+    email: string;
+    password: string;
+    avatar?: string;
+}
+
+const userSchema = new Schema<IUser>({
```

Comment:
Why did you use the Interface?

Reply:
I read it in the mongoose documentation: https://mongoosejs.com/docs/typescript/schemas.html
You can use automatic type inference to not specify the interface, but that has some downsides and I wanted to avoid that

Reply:
Ok also read through it - good call!